src/java.base/share/classes/java/text/CompactNumberFormat.java

```diff
@@ -47,112 +47,150 @@
 /**
  * <p>
  * {@code CompactNumberFormat} is a concrete subclass of {@code NumberFormat}
- * that formats a decimal number in its compact form.
- *
- * The compact number formatting is designed for the environment where the space
- * is limited, and the formatted string can be displayed in that limited space.
- * It is defined by LDML's specification for
+ * that formats a decimal number in a localized compact form.
+ * Compact number formatting is designed for an environment with limited space.
+ * For example, displaying the formatted number {@code 7M} instead of {@code
+ * 7,000,000.00} in the {@link java.util.Locale#US US locale}. The {@code
+ * CompactNumberFormat} class is defined by LDML's specification for
  * <a href = "http://unicode.org/reports/tr35/tr35-numbers.html#Compact_Number_Formats">
- * Compact Number Formats</a>. A compact number formatting refers
- * to the representation of a number in a shorter form, based on the patterns
- * provided for a given locale.
+ * Compact Number Formats</a>.
  *
- * <p>
- * For example:
- * <br>In the {@link java.util.Locale#US US locale}, {@code 1000} can be formatted
- * as {@code "1K"}, and {@code 1000000} as {@code "1M"}, depending upon the
- * {@linkplain ##compact_number_style style} used.
- * <br>In the {@code "hi_IN"} locale, {@code 1000} can be formatted as
- * "1 \u0939\u091C\u093C\u093E\u0930", and {@code 50000000} as "5 \u0915.",
- * depending upon the {@linkplain ##compact_number_style style} used.
+ * <h2>Getting a CompactNumberFormat</h2>
+ * To get a compact number format, use one of the ways listed below.
+ * <ul>
+ * <li> Use the factory method {@link NumberFormat#getCompactNumberInstance()}
+ * to obtain a format for the default locale with
+ * {@link NumberFormat.Style#SHORT SHORT} style.
+ * <li> Use the factory methood {@link NumberFormat#getCompactNumberInstance(Locale, Style)}
+ * to obtain a format for a different locale
+ * and to control the {@linkplain ##compact_number_style Style}.
+ * <li> Use one of the {@code CompactNumberFormat} constructors, for example, {@link
+ * CompactNumberFormat#CompactNumberFormat(String, DecimalFormatSymbols, String[])
+ * CompactNumberFormat(decimalPattern, symbols, compactPatterns)}, to obtain a
+ * {@code CompactNumberFormat} with further customization.
+ * </ul>
+ * <p>If a standard compact format for a given locale and {@link
+ * ##compact_number_style style} is desired, it is recommended to use one of the
+ * NumberFormat factory methods listed above. These factory methods may not always return
+ * a {@code CompactNumberFormat} depending on the locale-service provider implementation
+ * installed. Thus, to use an instance method defined by {@code CompactNumberFormat},
+ * the {@code NumberFormat} returned by the factory method should first be type
+ * checked before cast to {@code CompactNumberFormat}. If the installed locale-sensitive
```

Comment:
Since `CompactNumberFormat` does not provide its own instance methods (i.e., instance methods are exactly the same as the parent NumberFormat), "instance methods defined by CompactNF" does not make much sense, which is different for `DecimalFormat`.

Reply:
`CompactNumberFormat` does define its own instance methods such as `setParseBigDecimal`, `setGroupingSize`, etc, so I think that this wording should still be included.

Reply:
My bad, you're correct. The confusion I had was probably this wording: "These factory methods may not always return a CompactNumberFormat". I believe it *does* return a CNF (in terms of `instanceof`), because as the factory methods names imply. However, in the case of DecimalFormat, the factory methods are simply, for example, "NumberFormat.getInstance()" and the implementation may return a different class instance, which does not even offer methods on DecimalFormat class. The difference needs to be described somehow IMO.